Rebaseline negative-overflow-002.html on iOS due to RTL bugs

fantasai opened the pull request:
#### 43c839024fcd771899b9e1ae637cce440518eb26
<pre>
Rebaseline negative-overflow-002.html on iOS due to RTL bugs
<a href="https://bugs.webkit.org/show_bug.cgi?id=269116">https://bugs.webkit.org/show_bug.cgi?id=269116</a>
<a href="https://rdar.apple.com/122678250">rdar://122678250</a>

Reviewed by Ryosuke Niwa.

Rebaselines negative-overflow-002.html to have various 15px differences on iOS.

These seem to be due to incorrect handling of lefhand scrollbars when
calculating the scrollable area; iOS gives different results because it
has righthand scrollbars even in RTL.

This is a follow-up to <a href="https://commits.webkit.org/273737@main">https://commits.webkit.org/273737@main</a>

* LayoutTests/platform/ios-wk2/imported/w3c/web-platform-tests/css/css-flexbox/negative-overflow-002-expected.txt: Added.

Canonical link: <a href="https://commits.webkit.org/275526@main">https://commits.webkit.org/275526@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/b3a19ee58fa239556505891b5d7bdfcebb021806

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/41770 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/20784 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/44152 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/44347 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/37863 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/23926 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/18115 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/34524 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/42344 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/17712 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/35969 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/15193 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/15403 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/36993 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/45741 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/37956 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/37315 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/41070 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/16585 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/13618 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/39512 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/18204 "Built successfully") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/9424 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/18262 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/17848 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

smfr commented on LayoutTests/platform/ios-wk2/imported/w3c/web-platform-tests/css/css-flexbox/negative-overflow-002-expected.txt:
```diff
@@ -0,0 +1,406 @@
+1
```
This is a pretty silly expected result. Can the test use a <div style="height:200%"> for the scrolled contents rather than a bunch of newlines?

fantasai replied:
It's actually a bunch of numbered boxes. The numbers help you understand the test: http://wpt.live/css/css-flexbox/negative-overflow-002.html

fantasai committed:
Rebaseline negative-overflow-002.html on iOS due to RTL bugs
https://bugs.webkit.org/show_bug.cgi?id=269116
rdar://122678250

Reviewed by Ryosuke Niwa.

Rebaselines negative-overflow-002.html to have various 15px differences on iOS.

These seem to be due to incorrect handling of lefhand scrollbars when
calculating the scrollable area; iOS gives different results because it
has righthand scrollbars even in RTL.

This is a follow-up to https://commits.webkit.org/273737@main

* LayoutTests/platform/ios-wk2/imported/w3c/web-platform-tests/css/css-flexbox/negative-overflow-002-expected.txt: Added.

Canonical link: https://commits.webkit.org/275526@main
